src/Compilers/CSharp/Portable/Lowering/SynthesizedMethodBaseSymbol.cs

```diff
@@ -41,6 +41,11 @@ internal abstract class SynthesizedMethodBaseSymbol : SourceMethodSymbol
                 returnsVoid: baseMethod.ReturnsVoid,
                 isExtensionMethod: false,
                 isMetadataVirtualIgnoringModifiers: false);
+
+            if (this.ReturnsByRefReadonly)
```

Comment:
>if (this.ReturnsByRefReadonly) [](start = 12, length = 30)

It feels like we should have detected the need for the attribute before we reached this point.